otherlibs/stdune/src/fpath.ml

```diff
@@ -1,6 +1,18 @@
 let is_root t = Filename.dirname t = t
 
-let initial_cwd = Stdlib.Sys.getcwd ()
+let initial_cwd : string option =
+  try Some (Stdlib.Sys.getcwd ()) with _exn -> None
```

Comment:
Could you include the original exception in the error message? It should be `Sys_error msg` I believe.

It's possible that one couldn't determine the cwd because of permission issues perhaps. It would be useful to diagnose that if the actual error was included.

Reply:
Indeed, good idea.